### PR TITLE
Pending state while waiting for a convo to resolve

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this partially resurrects the idea of a pending waiting conversation but makes it much looser / simpler.
We already had a sentry value for this in constants so we just set it when we're going to resolve a conversation and select it. In that mode we show a simple 'loading...' text so it doesn't get annoying when if it flashes
Unlike before we don't do any kind of mapping between pending convos and resolving them, we just select the convo when its done and if we're looking at the pending we replace it

### CODE REVIEW

File: shared/actions/chat2/index.js [hunk 1]
```diff
@@ -1568,10 +1568,19 @@ const confirmScreenResponse = (_, action) => {
...
 
 // We always make adhoc convos and never preview it
-const previewConversationPersonMakesAConversation = (state, action) =>
-  !action.payload.teamname &&
-  action.payload.participants &&
-  Chat2Gen.createCreateConversation({participants: action.payload.participants})
+const previewConversationPersonMakesAConversation = (state, action) => {
+  const participants = action.payload.participants
+  return (
+    !action.payload.teamname &&
```
Comment (buoyad) [inline shared/actions/chat2/index.js]: :sheep: destructure with `participants`

File: shared/actions/chat2/index.js [hunk 2]
```diff
@@ -2064,8 +2077,20 @@ const navigateToInbox = (state, action, logger) => {
...
+  let replace = false
+
+  const visible = Router2Constants.getVisibleScreen()
+  // looking at the pending screen?
+  if (
+    visible?.routeName === 'chatConversation' &&
+    visible?.params?.conversationIDKey === Constants.pendingWaitingConversationIDKey
+  ) {
+    replace = true
+  }
```
Comment (buoyad) [inline shared/actions/chat2/index.js]: oh, a back button fix here regressed with deprecating nav1
https://github.com/keybase/client/pull/17439/files#diff-62d6b6a8baeb0bb4a699ed1965bcbf00R2110

File: shared/chat/conversation/messages/special-top-message.js [hunk 3]
```diff
@@ -79,6 +91,7 @@ type OwnProps = {
 const mapStateToProps = (state, ownProps: OwnProps) => {
   const hasLoadedEver = state.chat2.messageOrdinals.get(ownProps.conversationIDKey) !== undefined
   const meta = Constants.getMeta(state, ownProps.conversationIDKey)
+  const waitingPending = ownProps.conversationIDKey === Constants.pendingWaitingConversationIDKey
```
Comment (buoyad) [inline shared/chat/conversation/messages/special-top-message.js]: :sheep: `pendingWaiting` for consistency

File: shared/actions/chat2/index.js (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/chat2/index.js]: fixed and pushed tnx

**Pull request merged**